src/Matcher/Pattern/Expander/Repeat.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace Coduo\PHPMatcher\Matcher\Pattern\Expander;
+
+use Coduo\PHPMatcher\Factory\SimpleFactory;
+use Coduo\PHPMatcher\Matcher;
+use Coduo\PHPMatcher\Matcher\Pattern\PatternExpander;
+use Coduo\ToString\StringConverter;
+
+final class Repeat implements PatternExpander
+{
+    const NAME = 'repeat';
+
+    /**
+     * @var null|string
+     */
+    private $error;
+
+    /**
+     * @var string
+     */
+    private $pattern;
+
+    /**
+     * @var bool
+     */
+    private $isStrict;
+
+    /**
+     * @var bool
+     */
+    private $isScalar;
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function is($name)
+    {
+        return self::NAME === $name;
+    }
+
+    /**
+     * @param $value
+     */
+    public function __construct($pattern, $isStrict = true)
+    {
+        if (!is_string($pattern)) {
+            throw new \InvalidArgumentException("Repeat pattern must be a string.");
+        }
+
+        $this->pattern = $pattern;
+        $this->isStrict = $isStrict;
+        $this->isScalar = true;
+
+        $json = json_decode($pattern, true);
+
+        if ($json !== null && json_last_error() === JSON_ERROR_NONE) {
+            $this->pattern = $json;
+            $this->isScalar = false;
+        }
+    }
+
+    /**
+     * @param $values
+     * @return bool
+     */
+    public function match($values)
+    {
+        if (!is_array($values)) {
+            $this->error = sprintf("Repeat expander require \"array\", got \"%s\".", new StringConverter($values));
+            return false;
+        }
+
+        $factory = new SimpleFactory();
+        $matcher = $factory->createMatcher();
+
+        if ($this->isScalar) {
+            return $this->matchScalar($values, $matcher);
+        }
+
+        return $this->matchJson($values, $matcher);
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getError()
+    {
+        return $this->error;
+    }
+
+    /**
+     * @param array $values
+     * @param Matcher $matcher
+     * @return bool
+     */
+    private function matchScalar(array $values, Matcher $matcher)
+    {
+        foreach ($values as $index => $value) {
+            $match = $matcher->match($value, $this->pattern);
+
+            if (!$match) {
+                $this->error = sprintf("Repeat expander, entry n�%d, find error : %s", $index, $matcher->getError());
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * @param array $values
+     * @param Matcher $matcher
+     * @return bool
+     */
+    private function matchJson(array $values, Matcher $matcher)
```

Comment:
We should maybe allow this method to be recursive to match nested objects ?

Reply:
```json
{
"offers": "@array@.repeat({\"subscription\": \"@array@.repeat(\"date\": \"@string@.isDateTime()\")\"})"
}
```

This case will work with actual implementation ?

Reply:
This case should work with the nested array.repeat syntax 'cause I just make a call to php-matcher factory. But you're right, maybe allowing nested objects would be more clever.

Reply:
This is not yet supported, right?

Reply:
Yes I'm sorry, didn't have the time to do it